src/Tasks/Microsoft.NET.Build.Tasks/ResolvePublishAssemblies.cs

```diff
@@ -83,7 +83,7 @@ protected override void ExecuteCore()
                 TaskItem item = new TaskItem(resolvedAssembly.SourcePath);
                 item.SetMetadata("DestinationSubPath", resolvedAssembly.DestinationSubPath);
                 item.SetMetadata("AssetType", resolvedAssembly.Asset.ToString().ToLower());
-                item.SetMetadata(MetadataKeys.PackageName, resolvedAssembly.Package.Id.ToString().ToLower());
+                item.SetMetadata(MetadataKeys.PackageName, resolvedAssembly.Package.Id.ToString());
```

Comment:
Package Ids are *supposed* to be case insensitive, so in theory it is fine to make this change.  If it does break something, that something should be fixed to use a case insensitive comparison.

I did a quick look at where this is used, and the only place I found was it [gets passed into `FilterResolvedFiles`](https://github.com/dotnet/sdk/blob/cd25d6bbfc2aac61cc790c511900427ac64fdaa8/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeStore.targets#L426-L427), which then uses it to create a [PackageIdentity](https://github.com/NuGet/NuGet.Client/blob/1a1c474aab5c1e851b416c583ca4c6d727bd94e3/src/NuGet.Core/NuGet.Packaging.Core/PackageIdentity.cs#L13), which does case insensitive comparisons.  So this should be fine.